keras_core/ops/math.py

```diff
@@ -100,11 +101,12 @@ def segment_max(data, segment_ids, num_segments=None, sorted=False):
 
     Example:
 
-    >>> data = keras_core.ops.convert_to_tensor([1, 2, 3, 4, 5, 6])
-    >>> segment_ids = keras_core.ops.convert_to_tensor([0, 1, 0, 1, 0, 1])
-    >>> segment_max(data, segment_ids)
-    array([9 12], shape=(2,), dtype=int32)
-    """
+    >>> data = keras_core.ops.convert_to_tensor([1,2,10,20,100,200])
```

Comment:
Likewise here?

Reply:
1st , The issue was that the example does not work with the JAX backend.
``ValueError: Argument 'num_segments' must be set when using the JAX backend. Received: num_segments=None
`
The 2nd issue was that the wrong output ( I seems like it was a copy from segment_sum)

So, I have changed it to an easier-to-understand example, and now it is working with all backends.